seqbank/seqbank.py

```diff
@@ -367,4 +369,38 @@ def export(self, output:Path|str, format:str="", accessions:List[str]|str|Path|N
             for accession in accessions:
                 SeqIO.write(self.record(accession), f, format)
 
-    
\ No newline at end of file
+    def get_accession_lengths(self) -> dict:
+        """
+        Returns a dictionary where the keys are the accessions and the values 
+        are the corresponding lengths of each sequence.
+        """
+        accession_lengths = {}
+
+        for accession in self.get_accessions():
+            # Retrieve the sequence string using the string method
+            sequence = self.string(accession)
```

Comment:
instead of converting to string. Just access the data with:
data = self[accession]
accession_lengths[accession] = len(data)